pgjdbc/src/main/java/org/postgresql/jdbc/TimestampUtils.java

```diff
@@ -638,6 +638,12 @@ private static void appendTime(StringBuilder sb, int hours, int minutes, int sec
     if (needZeros > 0) {
       sb.insert(len, ZEROS, 0, needZeros);
     }
+
+    int end = sb.length() - 1;
+    while (sb.charAt(end) == '0') {
+      sb.deleteCharAt(end);
+      end--;
```

Comment:
Please add a test for `00.0` kind of timestamp.
I wonder if it should get all the zeros trimmed or if one should be kept.

PS. I somehow like `for(...)`  better